### ISSUE TITLE
🐛 BUG: Getting Error [36:29]: Expected ')' when I cast inside the getStaticPaths() function

### ISSUE THREAD

**Issue by mainawycliffe** [snowpackjs/astro#1594]
### What package manager are you using?

yarn

### What operating system are you using?

Ubuntu  21.04

### Describe the Bug

I am using a GhostCMS as my headless backend for my blog, I can successfully retrieve the list of posts from the API but when I try to type cast the data inside the getStaticPaths, it throws the error below, if I remove the casting section, it works smoothly.

```sh
[access] /blog/better-types-using-discriminated-types-in-typescript
[parse-error] [36:29]: Expected ')'
[access] /500
```

Here is what my function looks:

```ts
import { PostsOrPages } from "@tryghost/content-api";

export async function getStaticPaths() {
  const url = `URL_TO_LIST_BLOG_POSTS`;
  const response = await fetch(url);
  const res = await response.json();
  const posts = res.posts as PostsOrPages;
  return posts.map((post) => ({
    params: { slug: post.slug },
  }));
}
```

The casting works outside the function when I am retrieving the content of each page:

```ts
const { slug } = Astro.request.params;

// go ahead and fetch the post
const request = await fetch(
  `URL_TO_BLOG_POST`
);
const res = await request.json();
// casting here works
const content = (res as { posts: PostsOrPages }).posts[0];
```

### Steps to Reproduce

1. `npm init astro` 
2. add a getStaticPaths Method
3. cast the results from the source of an API using fetch
4. Error! Describe what went wrong (and what was expected instead)... should work but throws error: `[parse-error] [36:29]: Expected ')'`

**Comment by matthewp** [comment]
This seems to be working in `next`, marking this to be completed when that is released: https://stackblitz.com/edit/github-xzojkq?file=src/pages/posts/[...page].astro

**Comment by natemoo-re** [comment]
Following up on my previous comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
